src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -115,10 +115,10 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 			ArchiveFileList files = new ArchiveFileList ();
 			if (apkInputPath != null)
 				File.Copy (apkInputPath, temp, overwrite: true);
+			using (var notice = Assembly.GetExecutingAssembly ().GetManifestResourceStream ("NOTICE.txt"))
```

Comment:
It would be good measure to add a `.Flush()` call before the `using` block is exited as `Dispose()` doesn't flush buffers for some streams (neither does `Close()`)

Reply:
In this case we have our own `Flush` method (that isn't the real .NET `Flush`) that just does this: https://github.com/xamarin/xamarin-android/blob/936459c00e4a2da08e20cdbd83e5e23823529fbe/src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs#L30-L38

`ZipArchiveEx` does the same thing in `Dispose`: https://github.com/xamarin/xamarin-android/blob/936459c00e4a2da08e20cdbd83e5e23823529fbe/src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs#L145-L153